crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -100,13 +100,19 @@ impl RuffSettings {
 }
 
 impl RuffSettingsIndex {
-    pub(super) fn new(root: &Path, editor_settings: &ResolvedEditorSettings) -> Self {
+    pub(super) fn new(
+        root: &Path,
+        editor_settings: &ResolvedEditorSettings,
+        is_default_workspace: bool,
+    ) -> Self {
         let mut has_error = false;
         let mut index = BTreeMap::default();
         let mut respect_gitignore = None;
+        let should_skip_workspace = usize::from(!is_default_workspace);
 
-        // Add any settings from above the workspace root, excluding the workspace root itself.
-        for directory in root.ancestors().skip(1) {
+        // Add any settings from above the workspace root. The ones from the workspace root will be
+        // added only if it's not the default workspace.
```

Comment:
Nit: What interests me is why we're skipping the root itself if it is the default workspace. I think that would also make the comment easier to understand if it explains why we sometimes skip the root vs when we consider it.